File: go/01-variables/main_test.go
package main

import "testing"

func TestRobotName(t *testing.T) {
	if len(RobotName) == 0 {
		t.Errorf("RobotName must be defined")
	}
}

func TestBatteryLevel(t *testing.T) {
	if BatteryLevel != 98 {
		t.Errorf("got %d, want 98", BatteryLevel)
	}
}

func TestIsCharging(t *testing.T) {
	if IsCharging != false {
		t.Errorf("got %t, want false", IsCharging)
	}
}

File: rust/01-variables/tests/main_test.rs
use lesson_01_variables::*;

#[test]
fn test_robot_name() {
    assert_eq!(ROBOT_NAME, "R2D2");
}

#[test]
fn test_battery_level() {
    assert_eq!(BATTERY_LEVEL, 98);
}

#[test]
fn test_is_charging() {
    assert_eq!(IS_CHARGING, false);
}
